runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/CombineTranslation.java

```diff
@@ -242,6 +273,26 @@ public SdkFunctionSpec getCombineFn() {
       return accumulatorCoder;
     }
 
+    @Override
+    public Map<TupleTag<?>, PValue> getAdditionalInputs() {
+      // TODO: This was ripped from ParDoTranslation. Is this correct?
```

Comment:
Yes, this looks fine to me. Probably this could also be a helper that is shared between `ParDo` and `Combine`, since the canonical definition of how these side inputs should work is based on the composite definition of `Combine` as a GBK followed by a `ParDo`.

Reply:
Where should such a helper live?

Reply:
Any suggestions? Based on the size of this code, I don't think it's worth refactoring into a brand new class. I'll leave it here unless you have a better suggestion.